### ISSUE TITLE
implmentation of googlenet SSD

### ISSUE THREAD

**Issue by hengck23** [weiliu89/caffe#196]
I wonder if anyone try to use googlenet as feature extraction? I think this is very fast!

This link suggest that googlenet can be used for object detection too.
https://github.com/sanghoon/pva-faster-rcnn

I tried to use googlenet-bn from https://github.com/lim0606/caffe-googlenet-bn for SSD.
I can get speed of at about 22 ms for batch=1 at run time, however the  mAP is not high because:

- An object is detected multiple times (gives rise to false positive) because object parts are always detected. For example, an human detection can have 2 detection: the whole body and the head.
(I note that the ground truth in VOC do consists of different parts, e.g. head half body, full body, legs)

- all trees becomes potted plants!

Some examples of the images are shown below.  
How can I resolved such issues? Any help is much appreciated.

![000341 1](https://cloud.githubusercontent.com/assets/5112846/18921857/03856d72-85d9-11e6-855d-108b21cf6158.jpg)

![000076 1](https://cloud.githubusercontent.com/assets/5112846/18921856/0383abf4-85d9-11e6-8bc8-65291f5b9ca5.jpg)

**Comment by weiliu89** [comment]
I tried Inception-v3 a little bit. I could get about 69ish mAP. I converted the TF Inception-v3 model to caffe format, which might not be the right model (e.g. the top1 accuracy on ILSVRC val dataset is lower than the number it should have). I haven't investigated in more details.

The reason that it detects head as human might caused by the data augmentation during training as well.

**Comment by shawn-tian** [comment]
@hengck23 I'm not sure which layers to use for the mbox prediction. Could you able to share the prototxt of ssd on googlenet? thank you very much

**Comment by ngaloppo** [comment]
@weiliu89 Would you mind sharing your inception-v3 result? Also, have you considered inception-v4? You can generate caffe proto's with the code in this repo: https://github.com/soeaver/caffe-model

**Comment by weiliu89** [comment]
@ngaloppo I could share a model that I trained long time ago later, however I haven't really optimize for it. Besides, I converted a v3 TF model to caffe format, but I couldn't achieve the same accuracy it should have on CLS-LOC val dataset. Are there converted caffe models for inception v3 and v4?

**Comment by ngaloppo** [comment]
@weiliu89 Yes, the aforementioned https://github.com/soeaver/caffe-model includes models for inception v3 and v4.

**Comment by weiliu89** [comment]
@ngaloppo Are there any pretrained models? I only see the scripts defining the networks.

**Comment by CHUNYUWANG** [comment]
@hengck23 Can you share the prototxt file you used for training googlenet-ssd?

**Comment by jbmpark** [comment]
@weiliu89 @ngaloppo  I have much interests in your working job. Could you please share your pre-trained model and prototxt ?  
ps. I always appreciate all your hard contribution.

**Comment by soeaver** [comment]
@weiliu89 I release the the caffe model of inception-v3 and other networks, it may be helpful for you.
https://github.com/soeaver/caffe-model/tree/master/cls

**Comment by libra7** [comment]
@hengck23  Can you share the prototxt file you used for training googlenet-ssd? it may be helpful for us.

**Comment by guiyang882** [comment]
@weiliu89 @ngaloppo can you show me the pycaffe code to create the model about the googlenet or the inception model ? I want to modified those model to detect the small object.
thx !

**Comment by guiyang882** [comment]
I want to know how long have you train the inceptionv3 + SSD model.
